sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/data_collector.py

```diff
@@ -28,17 +31,19 @@ class DataCollector:
 
     def __init__(
         self,
-        enabled: Optional[bool] = None,
+        request: Optional[Request] = None,
+        response: Optional[Response] = None,
         rolling_rate: Optional[str] = None,
         destination: Optional[Destination] = None,
-        sampling_strategy: Optional[SamplingStrategy] = None,
```

Comment:
We're sure that we won't support other sampling strategies in the foreseeable future?

I guess it isn't a breaking change to support float|SamplingStrategy and we check the type. If float, then assume it's simple random sample. Else if SamplingStrategy we have sth more complex.

Ntd here, just a thought.